src/loaders.rs

```diff
@@ -79,16 +119,30 @@ impl FileSystemLoader {
     }
 }
 
-pub struct AppDirsLoader {}
+pub struct AppDirsLoader {
+    encoding: &'static Encoding,
+}
 
 impl AppDirsLoader {
+    pub fn new(encoding: &'static Encoding) -> Self {
+        Self { encoding }
+    }
+
     fn get_template(
         &self,
-        _py: Python<'_>,
-        _template_name: &str,
-        _engine: &EngineData,
+        py: Python<'_>,
+        template_name: &str,
+        engine: &EngineData,
     ) -> Result<PyResult<Template>, LoaderError> {
-        todo!()
+        let dirs_result = get_app_template_dirs(py, "templates");
+        let dirs = match dirs_result {
+            Ok(dirs) => dirs,
+            Err(e) => Vec::new(),
+        };
+
+        let filesystem_loader = FileSystemLoader::new(dirs, self.encoding);
+
+        filesystem_loader.get_template(py, template_name, engine)
```

Comment:
We can write this a bit more concisely with `unwrap_or_default()`:

```suggestion
        let dirs = get_app_template_dirs(py, "templates").unwrap_or_default();
        let filesystem_loader = FileSystemLoader::new(dirs, self.encoding);
        filesystem_loader.get_template(py, template_name, engine)
```

However, I wonder if we should be raising some or all of the errors from `get_app_template_dirs` up to the user:

```suggestion
        let dirs = get_app_template_dirs(py, "templates")?;
        let filesystem_loader = FileSystemLoader::new(dirs, self.encoding);
        filesystem_loader.get_template(py, template_name, engine)
```